lib/bech32/src/Codec/Binary/Bech32/Internal.hs

```diff
@@ -98,21 +101,21 @@ mkHumanReadablePart hrp
         Left HumanReadablePartTooShort
     | BS.length hrp > humanReadablePartMaxLength =
         Left HumanReadablePartTooLong
-    | BS.length invalidPortion > 0 =
-        Left $ HumanReadablePartContainsInvalidChar $ CharPosition $
-            BS.length validPortion
+    | not (null invalidCharPositions) =
+        Left $ HumanReadablePartContainsInvalidChars invalidCharPositions
     | otherwise =
         Right $ HumanReadablePart hrp
   where
-    (validPortion, invalidPortion) = BS.break (not . valid) hrp
+    invalidCharPositions = CharPosition . fst <$>
```

Comment:
This looks like a good candidate for `NonEmpty` instead of current `[]` . Additionally this would simplify L104 (and maybe others) as you would be guaranteed to have failure there

Reply:
> This looks like a good candidate for `NonEmpty` instead of current `[]` . Additionally this would simplify L104 (and maybe others) as you would be guaranteed to have failure there

Actually, it's possible for the `locateErrors` function to return the empty list, if it is not able to reliably determine the locations of errors (even if the string is provably erroneous).

I'm going to add a comment to the `locateErrors` function to make this clear.